src/js/common/schemaform/review/ReviewPage.jsx

```diff
@@ -45,6 +53,29 @@ class ReviewPage extends React.Component {
     }
   }
 
+  setPagesViewed = (keys) => {
+    const viewedPages = keys.reduce((pages, key) => {
+      if (!pages.has(key)) {
```

Comment:
Could we toggle a flag variable here rather than conducting the comparison on line 61?